profile/endpoints/endpoint-domain.go

```diff
@@ -88,53 +88,72 @@ func (ep *EndpointDomain) String() string {
 	return ep.renderPPP(ep.OriginalValue)
 }
 
+func isValidEndpointDomain(value string) bool {
+	// Check for root domain
+	if value == "." {
+		return true
+	}
+
+	// Check for a "contains" value.
+	if containsRegex.MatchString(value) {
+		return true
+	}
+
+	// Remove special leading characters.
+	value = strings.TrimLeft(value, ".*")
+
+	// Check if value is now a valid domain.
+	return netutils.IsValidFqdn(value)
+}
```

Comment:
This will fail if the domain can still have a `.*` suffix. The suffix is trimmed a few lines further down below after the call to `isValidEndpointDomain`

Also, the comments about a "contains" value and the "special leading characters" are a bit confusing